pkgs/development/python-modules/ipython-sql/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, ipython
+, ipython_genutils
+, prettytable
+, sqlalchemy
+, sqlparse
+}:
+buildPythonPackage rec {
+  pname = "ipython-sql";
+  version = "0.4.0";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-PoiOWb9XJ3y9bzg8sjKFiy18cSGeV0klcSjxbZhX5Gw=";
+  };
+
+  propagatedBuildInputs = [
+    ipython
+    ipython_genutils
+    prettytable
+    sqlalchemy
+    sqlparse
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py --replace 'prettytable<1' prettytable
+  '';
+
+  preBuild = ''
+    # setup.py reads this but it's not shipped with the PyPI package
+    touch NEWS.rst
+  '';
+
+  # tests aren't shipped with the PyPI package and there's no 0.4.0 tag in the
+  # repo
+  doCheck = false;
+
```

Comment:
I commented above on why fetching from GitHub isn't possible: there's no 0.4.0 tag.

Reply:
Ah OK.
I just had a look at the code and https://github.com/catherinedevlin/ipython-sql/commit/117764caf099d80100ed4b09fc004b55eed6f121 seems to be the 0.4.0 version commit.

Maybe it is worth using this instead of pypi?

Reply:
Fixed.